lesson_11/pop_victor/tema11-OOP3-VictorPOP.py

```diff
@@ -0,0 +1,73 @@
+from collections.abc import MutableMapping, Mapping, Iterable, Collection, Container, Sized
+
+
+class Student(object):
+
+    def __init__(self, first_name, last_name, age):
+        self.first_name = first_name
+        self.last_name = last_name
+        self.age = age
+
+    def __repr__(self):
+        class_name = type(self).__name__
+        return '{} ({} {} - {}) [{}]'.format(
+            class_name, self.first_name,
+            self.last_name, self.age, id(self)
+        )
+
+    def __str__(self):
+        return '{} {}, {}'.format(
+            self.first_name, self.last_name, self.age)
+
+
+class StudentsCollection():
+
+    def __init__(self, students_list=None):
+        self._students = list(students_list) if students_list else []
+
+    def __iter__(self):
+        return iter(self._students)
+
+    def __getitem__(self, index):
+        return self._students[index]
+
+    def __len__(self):
+        return len(self._students)
+
+    def __contains__(self, item):
+        for element in self._students:
+            print(str(element))
+            if item in str(element):
+                return True
+        return False
```

Comment:
I'm sure you can find a cleaner implementation for this ;)
It works, but there's definitely a better way

Reply:
the other way is commented below the code:
def __contains__(self, item):
    return item in str(self._students)

Reply:
or even 
def contains(self, item):
return item in str(self)

Reply:
yup